### PR TITLE
omniscidb 5.10.1 for modin

### PR DESCRIPTION
**Pull request by cbouss**
Repack of omniscidb-common, omniscidbe, pyomniscidbe for modin. modin 0.15.2 requires pyomniscidbe 5.10.1 with pyarrow==6.0.1.
The supported versions of arrow on default are 4 and 8. A decision was made to support pyomniscidbe 5.10.1 through a repack rather than adding full support of arrow 6 on default.

What to consider for a repack:
 - Tests results should be the same on both the original and repacked package.
 - Configure dependencies of repacked packages according to host run_export and run dependencies.
 - Isolate the repacked binaries in a dedicated folder.
 
 Changes:
- Repack of omniscidb-common, omniscidbe, pyomniscidbe
- Build of arrow-cpp 6.0.1 and pyarrow 6.0.1, which are included in the repack.
- Build of boost 1.74.0, also included in the repack.
- Commented test: test_exceptions.py. This test does not pass. This is also true in the original package.

Jira: https://anaconda.atlassian.net/browse/DSNC-4779

### CODE REVIEW

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -22,3 +22,9 @@ pin_run_as_build:
 docker_image:                                  # [linux64]
  - quay.io/condaforge/linux-anvil-comp7        # [linux64]
  - quay.io/condaforge/linux-anvil-cuda:11.0    # [linux64]
+
+# This needs to be removed after the pinning of libprotobuf has been adjusted.
+protobuf:
+- 3.20
```
Comment (katietz) [inline recipe/conda_build_config.yaml]: I would like to see it written as:
protobuf:
  - 3.20.1
... instead.  Intend is important

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -22,3 +22,9 @@ pin_run_as_build:
...
+libprotobuf:
+- 3.20
```
Comment (katietz) [inline recipe/conda_build_config.yaml]: same as above

**Review by katietz — APPROVED** [review]
with this change, package looks good to me. thank you

File: recipe/conda_build_config.yaml (hunk 1 above)
Comment (cbouss) [reply recipe/conda_build_config.yaml]: done, thank you

File: recipe/conda_build_config.yaml (hunk 2 above)
Comment (cbouss) [reply recipe/conda_build_config.yaml]: done, thank you

File: recipe/meta.yaml [hunk 3]
```diff
@@ -112,264 +122,171 @@ outputs:
...
-
+      license_file: omniscidb-common/share/doc/omnisci/LICENSE.md
+      summary: The OmniSci database common files.
       description: |
         OmniSciDB is an in-memory, column store, SQL relational database
         that was designed from the ground up to run on GPUs.
-
-        This recipe provides both CUDA enabled and CUDA disabled omniscidb
-        packages.
       doc_url: https://www.omnisci.com/docs/latest/
```
Comment (varlackc) [inline recipe/meta.yaml]: There is a redirect in the `doc_url` to the following `URL`
https://docs.heavy.ai/

File: recipe/meta.yaml [hunk 4]
```diff
@@ -112,264 +122,171 @@ outputs:
...
       dev_url: https://github.com/omnisci/omniscidb
```
Comment (varlackc) [inline recipe/meta.yaml]: There is a redirect in the `dev_url` to the following `URL`
https://github.com/heavyai/heavydb

File: recipe/meta.yaml [hunk 5]
```diff
@@ -383,91 +300,139 @@ outputs:
...
       license: Apache-2.0
       license_family: APACHE
-      license_file: LICENSE.md
+      license_file: omniscidb-common/share/doc/omnisci/LICENSE.md
       summary: The OmniSci database
-
       description: |
         OmniSciDB is an in-memory, column store, SQL relational database
         that was designed from the ground up to run on GPUs.
       doc_url: https://www.omnisci.com/docs/latest/
```
Comment (varlackc) [inline recipe/meta.yaml]: There is a redirect in the `doc_url` to the following `URL`
https://docs.heavy.ai/

File: recipe/meta.yaml [hunk 6]
```diff
@@ -383,91 +300,139 @@ outputs:
...
       dev_url: https://github.com/omnisci/omniscidb
```
Comment (varlackc) [inline recipe/meta.yaml]: There is a redirect in the `dev_url` to the following `URL`
https://github.com/heavyai/heavydb

File: recipe/meta.yaml [hunk 7]
```diff
@@ -476,49 +441,22 @@ outputs:
...
       license: Apache-2.0
       license_family: APACHE
-      license_file: LICENSE.md
+      license_file: omniscidb-common/share/doc/omnisci/LICENSE.md
       summary: The OmniSci database
-
       description: |
         OmniSciDB is an in-memory, column store, SQL relational database
         that was designed from the ground up to run on GPUs.
       doc_url: https://www.omnisci.com/docs/latest/
```
Comment (varlackc) [inline recipe/meta.yaml]: There is a redirect in the `doc_url` to the following `URL`
https://docs.heavy.ai/

File: recipe/meta.yaml [hunk 8]
```diff
@@ -476,49 +441,22 @@ outputs:
...
-  doc_url: https://www.omnisci.com/docs/latest/
-  dev_url: https://github.com/omnisci/omniscidb
-
-extra:
-  recipe-maintainers:
-    - xmnlab
-    - pearu
-    - andrewseidl
-    - jclay
+      dev_url: https://github.com/omnisci/omniscidb
```
Comment (varlackc) [inline recipe/meta.yaml]: There is a redirect in the `dev_url` to the following `URL`
https://github.com/heavyai/heavydb

File: recipe/meta.yaml [hunk 9]
```diff
@@ -383,91 +300,139 @@ outputs:
           source ${PREFIX}/etc/conda/activate.d/{{ PKG_NAME }}_activate.sh
           test -n "${OMNISCI_ROOT_PATH+x}"
           ldd ${PREFIX}/lib/libDBEngine.so
-
     about:
       home: https://www.omnisci.com/
```
Comment (varlackc) [inline recipe/meta.yaml]: There is a redirect in the `home URL`
https://www.heavy.ai/

File: recipe/meta.yaml [hunk 10]
```diff
@@ -476,49 +441,22 @@ outputs:
...
-        - python Embedded/test/test_readcsv.py
-        - pytest -sv Embedded/test/test_exceptions.py
-
+        - python source/Embedded/test/test_fsi.py
+        - python source/Embedded/test/test_readcsv.py
+        # Commenting test_exceptions as it crashes.
+        # The behavior is the same with the original pyomniscidbe package on conda-forge.
+        #- pytest -sv source/Embedded/test/test_exceptions.py
     about:
       home: https://www.omnisci.com/
```
Comment (varlackc) [inline recipe/meta.yaml]: There is a redirect in the `home URL`
https://www.heavy.ai/

File: recipe/meta.yaml [hunk 11]
```diff
@@ -112,264 +122,171 @@ outputs:
...
-            mkdir data && omnisci_initdb data
-            omnisci_server --version
-            rm -rf data
-          fi                                          # [cuda_compiler_version not in (undefined, "None")]
-        # these are provided by omniscidb-common package:
-        - which omnisci_insert_sample_data
-        - which startomnisci
-
     about:
       home: https://www.omnisci.com/
```
Comment (varlackc) [inline recipe/meta.yaml]: There is a redirect in the `home URL`
https://www.heavy.ai/

**Review by varlackc — COMMENTED** [review]
The `URLs` seem to have changed, other than that everything else looks good.

**Comment by cbouss** [comment]
> 

Thank you. I have updated the about section to match the rebranding to HeavyDB.

**Pull request merged**